src/ArchGDAL.jl

```diff
@@ -40,7 +40,7 @@ include("convert.jl")
 
 function __init__()
     GDAL.gdalallregister()
-    atexit(() -> GDAL.cplseterrorhandler(C_NULL))
+    atexit(() -> GDAL.gdaldestroydrivermanager())
```

Comment:
This can be simplified to atexit(GDAL.destroydrivermanager) since that is already a zero arg function.

Reply:
I overlooked it in the previous commit, but adding `() -> GDAL.gdaldestroydrivermanager()` with `atexit` in ArchGDAL brings back the issue "CPLDestroyMutex: Error = 16 (Resource busy)".

It may be linked to the fact that `atexit` adds functions in FIFO mode and that the custom error handler should be deregistered before calling `GDAL.gdaldestroydrivermanager()` 

If we want to keep `GDAL.gdaldestroydrivermanager()`, the two cleanup function should be called sequentially either in GDAL.jl or a client package as ArchGDAL

Reply:
Futhermore, as found [here](https://github.com/OSGeo/gdal/blob/3c518a6cac746b97f367578314e0bf784f78b861/gdal/apps/gdalasyncread.cpp#L558-L559) we could use `GDALDumpOpenDatasets(stderr)` to spot leaked dataset in tests

Reply:
The function `GDALDestroy` found [here](https://github.com/OSGeo/gdal/blob/3c518a6cac746b97f367578314e0bf784f78b861/gdal/gcore/gdaldllmain.cpp#L51-L100), hints to a more thorough cleanup and is called after `GDALDestroyDriverManager`call in some GDAL apps: ogr2ogr, gdalinfo, ogrinfo

Still, it seems that the custom error handler should not be called after as for `GDAL.gdaldestroydrivermanager()`

Reply:
I have tested a few combination. It seems that registering `gdaldestroy` at the beginning of GDAL.jl/gen/epilogue.jl `__init__()` function solves everything. We do not need to deregister the custom error handler in that same function, and since GDAL `GDALDestroy()` call `GDALDestroyDriverManager()`, we do not have to register its Julia wrapped version at exit.

It may be interresting to also register `() -> gdaldumpopendatasets("/dev/stderr")` after `gdaldestroy` to allow the user to identify unclosed datasets. But I have not found the way to pass a C pointer to `stderr` that works.

Shall I make a PR in GDAL.jl ?

Reply:
Well, I have made PR #124 in GDAL for comment.

And I have prepared, in case @visr accepts the latter PR, a commit below with no cleanup and but a few comments, in ArchGDAL.jl/src/ArchGDAL.jl

Reply:
> Well, I have made PR #124 in GDAL.jl for comments.

(I'm assuming you're referring to https://github.com/JuliaGeo/GDAL.jl/pull/124 -- github links that to https://github.com/yeesian/ArchGDAL.jl/issues/124 instead)

Sounds good to me. We'll need to wait for that PR to be merged and a new version of GDAL released, before updating the dependency on GDAL in https://github.com/yeesian/ArchGDAL.jl/blob/8f820c77c790d6166f03f17f145a4ffe474933de/Project.toml#L23

Reply:
Yes, I was referring to JuliaGeo/GDAL.jl#124

I had not seen that GDAL dependencies was specified up to the sub sub version. You are quite conservative :-)
I was using the master version when testing locally this PR and the related ArchGDAL

Reply:
All sloppiness traces to me, and all good credit goes to @visr really :)

Reply:
Haha not at all! @mathieu17g note that this compatability allows all 1.y.z GDAL.jl versions. The reason we used a patch version had to do with compatibility of the underlying build, see 47cc5564d23a7cde0591311908270444bc60643a.